### ISSUE TITLE
Maximum call stack size exceeded  RangeError: Maximum call stack size exceeded

### ISSUE THREAD

**Issue by Vykuntarao-n** [ArrayKnight/storybook-addon-headless#25]
**Describe the bug**

exceeded

    at http://localhost:9009/vendors~main.3b85dbc488b5875e321d.bundle.js:324288:52

    at Array.reduce (<anonymous>)

    at sortObject (http://localhost:9009/vendors~main.3b85dbc488b5875e321d.bundle.js:324288:36)

    at http://localhost:9009/vendors~main.3b85dbc488b5875e321d.bundle.js:324298:21

    at Array.reduce (<anonymous>)

    at sortObject (http://localhost:9009/vendors~main.3b85dbc488b5875e321d.bundle.js:324288:36)

    at http://localhost:9009/vendors~main.3b85dbc488b5875e321d.bundle.js:324298:21

    at Array.reduce (<anonymous>)

    at sortObject (http://localhost:9009/vendors~main.3b85dbc488b5875e321d.bundle.js:324288:36)

    at http://localhost:9009/vendors~main.3b85dbc488b5875e321d.bundle.js:324298:21

**To Reproduce**
 Please refer the environment details from my other tickets
I am using the storybook-addon-headless 2.1.1 the latest version, after starting the application I am getting this error

**Expected behavior**
Application should start with out any issue.

**Comment by ArrayKnight** [comment]
I can't tell from just this what the cause of the issue is. This doesn't appear to be occurring in a fresh install or in the examples that I've provided.

While none of my code directly implements a `sortObject` method, it's possible that one of my dependencies does

In order to be able to help I need to be able to reproduce the issue. If you can create an isolated environment with as few things as possible (no other addons, one component, minimal other dependencies, etc) I'm happy to take a look

**Comment by ArrayKnight** [comment]
@Vykuntarao-n were you able to resolve or reproduce the issue?

**Comment by binh710** [comment]
Hey @ArrayKnight , can you help to look into this issue again?

If I use this **config.ts** and **addons.ts** file when running storybook, it's fine
![image](https://user-images.githubusercontent.com/79820647/120135074-0dab8600-c202-11eb-9f5f-c8b236db5221.png)


But when I use **main.ts** file below
![image](https://user-images.githubusercontent.com/79820647/120135179-451a3280-c202-11eb-8d29-9d1dd5d6101b.png)

then I will see the "Maximum call stack size exceeded" error
![image](https://user-images.githubusercontent.com/79820647/120135204-52cfb800-c202-11eb-9cfc-a36a88d223ca.png)


Our goal here is to build a POC where we can call api within the mdx file. So unfortunately, I have to use the **main.ts** file configuration


Packages dependency info
```
"@storybook/addon-docs": "^6.2.9",
"@storybook/react": "^6.2.9",
"@types/storybook__react": "^5.2.1",
"storybook-addon-headless": "^2.1.1"
"react": "^17.0.2",
"react-dom": "^17.0.2",
"react-scripts": "4.0.3",
```

let me know if you need other information.

**Comment by binh710** [comment]
Hey @ArrayKnight , 

I would like to give you more snippet of code in case you need.


This is the code in **Users.tsx**

```
import React, { ReactElement } from "react"

export interface IPropsUsers {
    users: UserProps[]
}

export const User = (props: UserProps) => {
    const {
        id,
        name,
        email,
        website,
        company
    } = props;

    return <div>
        <h3>id: {id}</h3>
        <p>name: {name}</p>
        <p>email: {email}</p>
        <p>website: {website}</p>
        <p>company: {company?.name}</p>
    </div>
}

export const Users = (props: IPropsUsers): ReactElement | null => {
    const { users } = props;
    console.log('Userdata', users)

    return <>
        { users && users.length ? users.map((user) => 
            <User
                key={user.id}
                id={user.id}
                name={user.name}
                email={user.email}
                website={user.website}
                company={user.company}
            />     
        ) : null}
    </>
}


export interface UserProps {
    id: number
    name: string
    email: string
    website: string
    company: {
        name: string
    }
}
```


And this is the code in **Users.stories.tsx**

```
import { Args } from '@storybook/addons'
import React from "react";
import { storiesOf } from "@storybook/react";
import { Users } from "./Users";
import { 
    FetchStatus,
    HeadlessStoryContext,
    Loader,
    Prompt,
    withHeadless,
 } from 'storybook-addon-headless'

storiesOf("Users", module)
    .addDecorator(
        withHeadless({
            restful: {
                baseURL: 'https://jsonplaceholder.typicode.com/',
            },
        })
    )
    .addParameters({
        headless: {
            Users: {
                query: 'users',
                autoFetchOnInit: true,
            }
        }
    })
    .add("Users",
        (
            args: Args,
            { status, data }
        ) => {
            switch (status?.Users) {
                case FetchStatus.Inactive:
                case FetchStatus.Rejected:
                    return <Prompt />
        
                case FetchStatus.Loading:
                    return <Loader />
        
                default:
                    return <Users users={data?.Users} />
            }
        }
    )
```

I tried to convert **Users.stories.tsx** to this format, but I still see the same error

```
import { Args } from '@storybook/addons'
import React, { ReactElement } from "react";
import { storiesOf } from "@storybook/react";
import { Users } from "./Users";
import { 
    FetchStatus,
    HeadlessStoryContext,
    Loader,
    Prompt,
    withHeadless,
 } from 'storybook-addon-headless'

export default {
    title: 'Users',
    decorators: [
        withHeadless({
            restful: {
                baseURL: 'https://jsonplaceholder.typicode.com/',
            },
        }),
    ],
    parameters: {
        headless: {
            Users: {
                query: 'users',
                autoFetchOnInit: true,
            }
        }
    }
}

export const UsersStory = (
    args: Args,
    { status, data }
): ReactElement => {
    console.log("status", status);
    console.log("data", data);

    switch (status?.Users) {
        case FetchStatus.Inactive:
        case FetchStatus.Rejected:
            return <Prompt />

        case FetchStatus.Loading:
            return <Loader />

        default:
            return Array.isArray(data?.Users) ? (
                <Users users={data?.Users} />
            ) : <div>Unexpected Error</div>
    }
}
```

**Comment by ArrayKnight** [comment]
You'll have to give me a little time to get to this. I'm pretty busy right now. But in general I'd also like to ask what Node version you're on and which package manager (NPM, Yarn, PNPM, etc) you're using and version.

Also, is this occurring on an entirely new fresh project with just the code you've provided me? Or were you adding Storybook to an existing project? Just trying to isolate the issue as much as possible

**Comment by binh710** [comment]
Heyya, 
Thanks for getting back to me so quickly.

Node version: **v15.12.0**
I m using **yarn**
It's a POC so I start the project from scratch using **create-react-app**

**Comment by ArrayKnight** [comment]
Alright, I've managed to setup a quick project that matches yours. It appears that this only occurs when you also include `@storybook/addon-docs`, even though you're not actually using it so far.

By removing this addon the error doesn't occur. Obviously, this isn't ideal but it also means there's a lot more to dive into here.

Is this a problem with my addon? Is this a problem with `@storybook/addon-docs`? Or is there a conflict within Storybook when multiple addons with certain features are added together?

As another quick test, I removed `@storybook/addon-docs` and added  `@storybook/addon-essentials` which also implements it, same issue.

I also noticed that the addons actually appear to be functioning, it is simply the story that is erroring. Hmm...
